lightning/src/ln/channel.rs

```diff
@@ -7252,6 +7777,514 @@ impl<SP: Deref> InboundV1Channel<SP> where SP::Target: SignerProvider {
+		let keys = self.context.get_holder_pubkeys();
+
+		msgs::OpenChannelV2 {
+			chain_hash,
+			temporary_channel_id: self.context.temporary_channel_id.unwrap(),
+			funding_satoshis: self.context.channel_value_satoshis,
+			dust_limit_satoshis: self.context.holder_dust_limit_satoshis,
+			max_htlc_value_in_flight_msat: self.context.holder_max_htlc_value_in_flight_msat,
+			htlc_minimum_msat: self.context.holder_htlc_minimum_msat,
+			funding_feerate_sat_per_1000_weight: self.context.feerate_per_kw,
+			commitment_feerate_sat_per_1000_weight: self.context.feerate_per_kw,
+			to_self_delay: self.context.get_holder_selected_contest_delay(),
+			max_accepted_htlcs: self.context.holder_max_accepted_htlcs,
+			funding_pubkey: keys.funding_pubkey,
+			revocation_basepoint: keys.revocation_basepoint.to_public_key(),
+			payment_basepoint: keys.payment_point,
+			delayed_payment_basepoint: keys.delayed_payment_basepoint.to_public_key(),
+			htlc_basepoint: keys.htlc_basepoint.to_public_key(),
+			first_per_commitment_point,
+			second_per_commitment_point,
+			channel_flags: if self.context.config.announced_channel {1} else {0},
+			shutdown_scriptpubkey: Some(match &self.context.shutdown_scriptpubkey {
+				Some(script) => script.clone().into_inner(),
+				None => Builder::new().into_script(),
+			}),
+			channel_type: Some(self.context.channel_type.clone()),
+			locktime: self.dual_funding_context.funding_tx_locktime.to_consensus_u32(),
+			require_confirmed_inputs: None,
+		}
+	}
+
+	pub fn begin_interactive_funding_tx_construction<ES: Deref>(&mut self, signer_provider: &SP,
+		entropy_source: &ES, holder_node_id: PublicKey, funding_inputs: Vec<(TxIn, Transaction)>
+	) -> Result<Option<InteractiveTxMessageSend>, APIError>
+	where ES::Target: EntropySource
+	{
+		self.context.begin_interactive_funding_tx_construction(&self.dual_funding_context,
+			signer_provider, entropy_source, holder_node_id, true /* is_initiator */, funding_inputs)
+	}
+
+	pub fn funding_tx_constructed<L: Deref>(
+		mut self, counterparty_node_id: &PublicKey, signing_session: InteractiveTxSigningSession, signer_provider: &SP, logger: &L
+	) -> Result<(Channel<SP>, msgs::CommitmentSigned, Option<Event>), (Self, ChannelError)>
+	where
+		L::Target: Logger
+	{
+		let mut output_index = None;
+		let expected_spk = self.context.get_funding_redeemscript().to_v0_p2wsh();
+		for (idx, outp) in signing_session.constructed_transaction.output.iter().enumerate() {
+			if outp.script_pubkey == expected_spk && outp.value == self.context.get_value_satoshis() {
+				if output_index.is_some() {
+					return Err((self, ChannelError::Close("Multiple outputs matched the expected script and value".to_owned())));
+				}
+				output_index = Some(idx as u16);
+			}
+		}
+		if output_index.is_none() {
+			if output_index.is_some() {
+				return Err((self, ChannelError::Close("No output matched the script_pubkey and value in the FundingGenerationReady event".to_owned())));
+			}
+		}
```

Comment:
```suggestion
		if output_index.is_none() {
			return Err((self, ChannelError::Close("No output matched the script_pubkey and value in the FundingGenerationReady event".to_owned())));
		}
```

Statement will never execute, looks like a typo.